### PR TITLE
Support failOnStdErr on Helm Task

### PR DESCRIPTION
**Pull request by balteravishay**
Fixes issue [1660207](https://dev.azure.com/mseng/AzureDevOps/_workitems/edit/1660207/): In Helm3 some of the commands writing output to stderr due to this even deployment successful, task showing fail because task look at stderr and if any data exist it fails

Suggested fix: 

* Adding an option to not fail on standard error. 
* A tick mark (true by default) that adds Fail on Standard Error.
* customers can choose if they don't want to fail on std error.

### CODE REVIEW

File: Tasks/HelmDeployV0/task.json [hunk 1]
```diff
@@ -383,6 +383,14 @@
             "helpMarkDown": "Specify K8 namespace of tiller.",
             "type": "string",
             "groupName": "advanced"
+        },
+        {
+            "name": "failOnStderr",
+            "type": "boolean",
+            "label": "Fail on Standard Error",
+            "defaultValue": "true",
```
Comment (vithati) [inline Tasks/HelmDeployV0/task.json]: Keep default false else it will break compat.

File: Tasks/HelmDeployV0/src/helm.ts [hunk 2]
```diff
@@ -73,6 +73,7 @@ async function run() {
         command: command,
         jobId: tl.getVariable('SYSTEM_JOBID')
     };
+    var failOnStderr = tl.getBoolInput("failOnStderr", true);
```
Comment (vithati) [inline Tasks/HelmDeployV0/src/helm.ts]: it is not mandatory input so keep false.

Increase task.json minor version to 165 and pathversion to 0.

Comment (balteravishay) [reply Tasks/HelmDeployV0/src/helm.ts]: fixed both

File: Tasks/HelmDeployV0/task.json [hunk 3]
```diff
@@ -13,7 +13,7 @@
     "author": "Microsoft Corporation",
     "version": {
         "Major": 0,
-        "Minor": 165,
+        "Minor": 166,
```
[the remark this answers is not in the document]
Comment (balteravishay) [reply Tasks/HelmDeployV0/task.json]: fixed

**Pull request merged**